src/System.IdentityModel.Tokens.Jwt/JwtSecurityTokenHandler.cs

```diff
@@ -382,7 +383,31 @@ public virtual string CreateEncodedJwt(string issuer, string audience, ClaimsIde
         /// <exception cref="ArgumentException">If 'expires' &lt;= 'notBefore'.</exception>
         public virtual string CreateEncodedJwt(string issuer, string audience, ClaimsIdentity subject, DateTime? notBefore, DateTime? expires, DateTime? issuedAt, SigningCredentials signingCredentials, EncryptingCredentials encryptingCredentials)
         {
-            return CreateJwtSecurityTokenPrivate(issuer, audience, subject, notBefore, expires, issuedAt, signingCredentials, encryptingCredentials).RawData;
+            return CreateJwtSecurityTokenPrivate(issuer, audience, subject, notBefore, expires, issuedAt, signingCredentials, encryptingCredentials, null).RawData;
+        }
+
+        /// <summary>
+        /// Creates a JWT in 'Compact Serialization Format'.
+        /// </summary>
+        /// <param name="issuer">The issuer of the token.</param>
+        /// <param name="audience">The audience for this token.</param>
+        /// <param name="subject">The source of the <see cref="Claim"/>(s) for this token.</param>
+        /// <param name="notBefore">Translated into 'epoch time' and assigned to 'nbf'.</param>
+        /// <param name="expires">Translated into 'epoch time' and assigned to 'exp'.</param>
+        /// <param name="issuedAt">Translated into 'epoch time' and assigned to 'iat'.</param>
+        /// <param name="signingCredentials">Contains cryptographic material for signing.</param>
+        /// <param name="encryptingCredentials">Contains cryptographic material for encrypting.</param>
+        /// <param name="claimCollection">A collection of (key,value) pairs representing <see cref="Claim"/>(s) for this token.</param>
+        /// <remarks>If <see cref="ClaimsIdentity.Actor"/> is not null, then a claim { actort, 'value' } will be added to the payload. <see cref="CreateActorValue"/> for details on how the value is created.
+        /// <para>See <seealso cref="JwtHeader"/> for details on how the HeaderParameters are added to the header.</para>
+        /// <para>See <seealso cref="JwtPayload"/> for details on how the values are added to the payload.</para>
+        /// <para>Each <see cref="Claim"/> in the <paramref name="subject"/> will map <see cref="Claim.Type"/> by applying <see cref="OutboundClaimTypeMap"/>. Modifying <see cref="OutboundClaimTypeMap"/> could change the outbound JWT.</para>
+        /// </remarks>
+        /// <returns>A Base64UrlEncoded string in 'Compact Serialization Format'.</returns>
+        /// <exception cref="ArgumentException">If 'expires' &lt;= 'notBefore'.</exception>
+        public virtual string CreateEncodedJwt(string issuer, string audience, ClaimsIdentity subject, DateTime? notBefore, DateTime? expires, DateTime? issuedAt, SigningCredentials signingCredentials, EncryptingCredentials encryptingCredentials, IDictionary<string, object> claimCollection)
```

Comment:
Do we need this public method? It a bit confusing to me. Could a user accomplish the same by calling CreateEncodedJwt(SecurityTokenDescriptor)?

Reply:
Yes, accomplishes same as CreateEncodedJwt(SecurityTokenDescriptor).

Before this change [this](https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/e4928aff114bf17ee04bd0360b15da6d33694b06/src/System.IdentityModel.Tokens.Jwt/JwtSecurityTokenHandler.cs#L383) was also acheiving the same as CreateEncodedJwt(SecurityTokenDescriptor) ? Please correct me if I am wrong.